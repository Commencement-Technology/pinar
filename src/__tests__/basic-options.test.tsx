import { shallow } from "enzyme";
import React from "react";
import { Text, View } from "react-native";
import Carousel from "../index";

jest.mock("Dimensions");

describe("basic options", (): void => {
  it("should render with default settings", (): void => {
    const wrapper = shallow(
      <Carousel>
        <View>
          <Text>1</Text>
        </View>
        <View>
          <Text>2</Text>
        </View>
      </Carousel>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should not show prev/next buttons when there is only one slide", (): void => {
    const wrapper = shallow(
      <Carousel>
        <View>
          <Text>1</Text>
        </View>
      </Carousel>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should support disabling prev/next controls", (): void => {
    const wrapper = shallow(
      <Carousel showsControls={false}>
        <View>
          <Text>1</Text>
        </View>
        <View>
          <Text>2</Text>
        </View>
      </Carousel>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should support custom a11y labels for prev/next controls", (): void => {
    const wrapper = shallow(
      <Carousel
        accessibilityLabelNext="custom next label"
        accessibilityLabelPrev="custom prev label"
        index={1}
      >
        <View>
          <Text>1</Text>
        </View>
        <View>
          <Text>2</Text>
        </View>
        <View>
          <Text>3</Text>
        </View>
      </Carousel>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should support disabling dots", (): void => {
    const wrapper = shallow(
      <Carousel showsDots={false}>
        <View>
          <Text>1</Text>
        </View>
        <View>
          <Text>2</Text>
        </View>
      </Carousel>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should support providing height and width", (): void => {
    const wrapper = shallow(
      <Carousel height={333} width={444}>
        <View>
          <Text>1</Text>
        </View>
        <View>
          <Text>2</Text>
        </View>
      </Carousel>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should support loop option", (): void => {
    /*
    Looping is implemented by modifying the pages:
    - without loop: 1-2-3
    - with loop: 3-1-2-3-1

    ...and by changing the logic of updating
    the page index and scroll offset to work
    differently when loop is enabled.
    */
    const wrapper = shallow(
      <Carousel loop={true}>
        <View>
          <Text>1</Text>
        </View>
        <View>
          <Text>2</Text>
        </View>
        <View>
          <Text>3</Text>
        </View>
      </Carousel>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should support disabling accessibility", (): void => {
    const wrapper = shallow(
      <Carousel accessibility={false} loop={true}>
        <View>
          <Text>1</Text>
        </View>
        <View>
          <Text>2</Text>
        </View>
      </Carousel>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
