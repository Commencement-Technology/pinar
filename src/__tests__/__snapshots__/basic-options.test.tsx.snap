// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic options should not show prev/next buttons when there is only one slide 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <ScrollView
      automaticallyAdjustContentInsets={false}
      bounces={false}
      contentOffset={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      horizontal={true}
      onMomentumScrollEnd={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollEnabled={true}
      scrollEventThrottle={16}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
    >
      <View
        accessible={true}
        key="0"
        style={
          Object {
            "height": 800,
            "width": 480,
          }
        }
      >
        <View
          key=".0"
        >
          <Text>
            1
          </Text>
        </View>
      </View>
    </ScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 25,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        key="0/.0"
        style={
          Object {
            "backgroundColor": "#333",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "left": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 800,
            "width": 480,
          },
        ]
      }
    >
      <View />
      <View />
    </View>
  </View>
</View>
`;

exports[`basic options should render with default settings 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <ScrollView
      automaticallyAdjustContentInsets={false}
      bounces={false}
      contentOffset={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      horizontal={true}
      onMomentumScrollEnd={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollEnabled={true}
      scrollEventThrottle={16}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
    >
      <View
        accessible={true}
        key="0"
        style={
          Object {
            "height": 800,
            "width": 480,
          }
        }
      >
        <View
          key=".0"
        >
          <Text>
            1
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        key="1"
        style={
          Object {
            "height": 800,
            "width": 480,
          }
        }
      >
        <View
          key=".1"
        >
          <Text>
            2
          </Text>
        </View>
      </View>
    </ScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 25,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        key="0/.0"
        style={
          Object {
            "backgroundColor": "#333",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
      <View
        key="1/.1"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "left": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 800,
            "width": 480,
          },
        ]
      }
    >
      <View />
      <TouchableOpacity
        accessibilityComponentType="button"
        accessibilityLabel="Next"
        accessibilityRole="button"
        accessibilityTraits="button"
        accessible={true}
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          accessibilityLabel="Next"
          accessible={true}
          style={
            Object {
              "color": "#333",
              "fontSize": 50,
              "padding": 20,
            }
          }
        >
          â€º
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;

exports[`basic options should support disabling accessibility 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <ScrollView
      automaticallyAdjustContentInsets={false}
      bounces={false}
      contentOffset={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      horizontal={true}
      onMomentumScrollEnd={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollEnabled={true}
      scrollEventThrottle={16}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
    >
      <View
        accessible={false}
        key="0"
        style={
          Object {
            "height": 800,
            "width": 480,
          }
        }
      >
        <View
          key=".1"
        >
          <Text>
            2
          </Text>
        </View>
      </View>
      <View
        accessible={false}
        key="1"
        style={
          Object {
            "height": 800,
            "width": 480,
          }
        }
      >
        <View
          key=".0"
        >
          <Text>
            1
          </Text>
        </View>
      </View>
      <View
        accessible={false}
        key="2"
        style={
          Object {
            "height": 800,
            "width": 480,
          }
        }
      >
        <View
          key=".1"
        >
          <Text>
            2
          </Text>
        </View>
      </View>
      <View
        accessible={false}
        key="3"
        style={
          Object {
            "height": 800,
            "width": 480,
          }
        }
      >
        <View
          key=".0"
        >
          <Text>
            1
          </Text>
        </View>
      </View>
    </ScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 25,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        key="0/.0"
        style={
          Object {
            "backgroundColor": "#333",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
      <View
        key="1/.1"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "left": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 800,
            "width": 480,
          },
        ]
      }
    >
      <TouchableOpacity
        accessibilityComponentType="button"
        accessibilityLabel="Previous"
        accessibilityRole="button"
        accessibilityTraits="button"
        accessible={false}
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          accessibilityLabel="Previous"
          accessible={false}
          style={
            Object {
              "color": "#333",
              "fontSize": 50,
              "padding": 20,
            }
          }
        >
          â€¹
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        accessibilityComponentType="button"
        accessibilityLabel="Next"
        accessibilityRole="button"
        accessibilityTraits="button"
        accessible={false}
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          accessibilityLabel="Next"
          accessible={false}
          style={
            Object {
              "color": "#333",
              "fontSize": 50,
              "padding": 20,
            }
          }
        >
          â€º
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;

exports[`basic options should support disabling dots 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <ScrollView
      automaticallyAdjustContentInsets={false}
      bounces={false}
      contentOffset={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      horizontal={true}
      onMomentumScrollEnd={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollEnabled={true}
      scrollEventThrottle={16}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
    >
      <View
        accessible={true}
        key="0"
        style={
          Object {
            "height": 800,
            "width": 480,
          }
        }
      >
        <View
          key=".0"
        >
          <Text>
            1
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        key="1"
        style={
          Object {
            "height": 800,
            "width": 480,
          }
        }
      >
        <View
          key=".1"
        >
          <Text>
            2
          </Text>
        </View>
      </View>
    </ScrollView>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "left": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 800,
            "width": 480,
          },
        ]
      }
    >
      <View />
      <TouchableOpacity
        accessibilityComponentType="button"
        accessibilityLabel="Next"
        accessibilityRole="button"
        accessibilityTraits="button"
        accessible={true}
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          accessibilityLabel="Next"
          accessible={true}
          style={
            Object {
              "color": "#333",
              "fontSize": 50,
              "padding": 20,
            }
          }
        >
          â€º
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;

exports[`basic options should support disabling prev/next controls 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <ScrollView
      automaticallyAdjustContentInsets={false}
      bounces={false}
      contentOffset={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      horizontal={true}
      onMomentumScrollEnd={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollEnabled={true}
      scrollEventThrottle={16}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
    >
      <View
        accessible={true}
        key="0"
        style={
          Object {
            "height": 800,
            "width": 480,
          }
        }
      >
        <View
          key=".0"
        >
          <Text>
            1
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        key="1"
        style={
          Object {
            "height": 800,
            "width": 480,
          }
        }
      >
        <View
          key=".1"
        >
          <Text>
            2
          </Text>
        </View>
      </View>
    </ScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 25,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        key="0/.0"
        style={
          Object {
            "backgroundColor": "#333",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
      <View
        key="1/.1"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`basic options should support loop option 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <ScrollView
      automaticallyAdjustContentInsets={false}
      bounces={false}
      contentOffset={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      horizontal={true}
      onMomentumScrollEnd={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollEnabled={true}
      scrollEventThrottle={16}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
    >
      <View
        accessible={true}
        key="0"
        style={
          Object {
            "height": 800,
            "width": 480,
          }
        }
      >
        <View
          key=".2"
        >
          <Text>
            3
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        key="1"
        style={
          Object {
            "height": 800,
            "width": 480,
          }
        }
      >
        <View
          key=".0"
        >
          <Text>
            1
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        key="2"
        style={
          Object {
            "height": 800,
            "width": 480,
          }
        }
      >
        <View
          key=".1"
        >
          <Text>
            2
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        key="3"
        style={
          Object {
            "height": 800,
            "width": 480,
          }
        }
      >
        <View
          key=".2"
        >
          <Text>
            3
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        key="4"
        style={
          Object {
            "height": 800,
            "width": 480,
          }
        }
      >
        <View
          key=".0"
        >
          <Text>
            1
          </Text>
        </View>
      </View>
    </ScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 25,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        key="0/.0"
        style={
          Object {
            "backgroundColor": "#333",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
      <View
        key="1/.1"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
      <View
        key="2/.2"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "left": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 800,
            "width": 480,
          },
        ]
      }
    >
      <TouchableOpacity
        accessibilityComponentType="button"
        accessibilityLabel="Previous"
        accessibilityRole="button"
        accessibilityTraits="button"
        accessible={true}
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          accessibilityLabel="Previous"
          accessible={true}
          style={
            Object {
              "color": "#333",
              "fontSize": 50,
              "padding": 20,
            }
          }
        >
          â€¹
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        accessibilityComponentType="button"
        accessibilityLabel="Next"
        accessibilityRole="button"
        accessibilityTraits="button"
        accessible={true}
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          accessibilityLabel="Next"
          accessible={true}
          style={
            Object {
              "color": "#333",
              "fontSize": 50,
              "padding": 20,
            }
          }
        >
          â€º
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;

exports[`basic options should support providing height and width 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "height": 333,
        "width": 444,
      }
    }
  >
    <ScrollView
      automaticallyAdjustContentInsets={false}
      bounces={false}
      contentOffset={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      horizontal={true}
      onMomentumScrollEnd={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollEnabled={true}
      scrollEventThrottle={16}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
    >
      <View
        accessible={true}
        key="0"
        style={
          Object {
            "height": 333,
            "width": 444,
          }
        }
      >
        <View
          key=".0"
        >
          <Text>
            1
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        key="1"
        style={
          Object {
            "height": 333,
            "width": 444,
          }
        }
      >
        <View
          key=".1"
        >
          <Text>
            2
          </Text>
        </View>
      </View>
    </ScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 25,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        key="0/.0"
        style={
          Object {
            "backgroundColor": "#333",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
      <View
        key="1/.1"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "left": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 333,
            "width": 444,
          },
        ]
      }
    >
      <View />
      <TouchableOpacity
        accessibilityComponentType="button"
        accessibilityLabel="Next"
        accessibilityRole="button"
        accessibilityTraits="button"
        accessible={true}
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          accessibilityLabel="Next"
          accessible={true}
          style={
            Object {
              "color": "#333",
              "fontSize": 50,
              "padding": 20,
            }
          }
        >
          â€º
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;
